tests/test_mutations.py

```diff
@@ -0,0 +1,245 @@
+        mut_seq3 = mutations._apply_mutations(ref_seq[25:], 25, 26, mut_dict)  # [25,26) not include 26
+        assert mut_seq3 == gt_mut_seq3[25:]
+
+    def test_get_mutated_sequence(self):
+        fasta_file = os.path.join(test_data_dir, 'small_genome.fa')
+
+        ref_seq = 'GTAATGTGTAAGATGACGCACGCATGGTGGTATTGGAGATGGGTTGCGGAGTAAGTTCGAGTTC'
+        mutated_seq = 'GTAATGTGTAAGATGACGCACGCAT' + 'AC' + 'TGGTATTGGAGATGGGTTGCGGAGTAAGTTCGAGTTC'
+        mut_dict = {}
+        mut_dict[9] = {'mut_base': 'G', 'ref_base': 'G'}  # outside range
+        mut_dict[20] = {'mut_base': '*', 'ref_base': 'A'}
+        mut_dict[35] = {'mut_base': 'A', 'ref_base': 'G'}
+        mut_dict[36] = {'mut_base': 'C', 'ref_base': 'G'}
+        mut_dict[75] = {'mut_base': 'T', 'ref_base': 'G'}  # outside range
+
+        # the sub-sequence read from the fasta file is ref_seq
+        seq_dict = mutations.get_mutated_sequence(fasta_file, 'chr1', 10, 74, mut_dict)  # test unclear mut_base
+        assert ref_seq == seq_dict['ref']
+        assert mutated_seq == seq_dict['background']
+
+    def test_get_mutated_sequence_beginning_and_end(self):
+        """ Test that mutations are correctly applied when they are right at the beginning and end """
+        fasta_file = os.path.join(test_data_dir, 'small_genome.fa')
+
+        ref_seq = 'GTAATGTGTAAGATGACGCACGCATGGTGGTATTGGAGATGGGTTGCGGAGTAAGTTCGAGTTC'
+        mutated_seq = 'A' + 'TAATGTGTAAGATGACGCACGCATGGTGGTATTGGAGATGGGTTGCGGAGTAAGTTCGAGTT' + 'G'
+        mut_dict = {}
+        mut_dict[10] = {'mut_base': 'A', 'ref_base': 'G'}
+        mut_dict[20] = {'mut_base': '*', 'ref_base': 'A'}
+        mut_dict[73] = {'mut_base': 'G', 'ref_base': 'G'}
+
+        # the sub-sequence read from the fasta file is ref_seq
+        seq_dict = mutations.get_mutated_sequence(fasta_file, 'chr1', 10, 74, mut_dict)  # test unclear mut_base
+        assert ref_seq == seq_dict['ref']
+        assert mutated_seq == seq_dict['background']
+
+
+class TestLoadMutations:
+    @pytest.fixture
+    def basic_args(self):
+        return ['build',
+                '--output-samples', 'sample1, sample2',
+                '--splice-path', 'this_splicegraph',
+                '--output-dir', 'this_output_dir',
+                '--ann-path', 'this_ann_path',
+                '--ref-path', 'this_ref_path']
+
+    def test_reference(self):
+        """ Tests that when using --build-mode ref, the mutation dicts are empty """
+        mutation = mutations.load_mutations(germline_file=None, somatic_file=None, mutation_sample='', heter_code=0,
+                                            target_samples=['sample1', 'sample2'], sample_name_map_file=None,
+                                            cache_dir=None)
+        assert mutation.somatic_dict == {}
+        assert mutation.germline_dict == {}
+
+    def test_germline(self, basic_args):
+        """ Tests that an error is thrown when requesting germline mutation mode, but --germline is missing. """
```

Comment:
These tests seam to contradict my assumption made further above that --mutation_mode is no longer used?